app/src/main/java/com/danosoftware/galaxyforce/services/googleplay/GooglePlayServices.java

```diff
@@ -0,0 +1,548 @@
+                        Log.i(ACTIVITY_TAG, "Success signedIn");
+                        onConnected(googleSignInAccount, ConnectionRequest.LOG_IN);
+                    }
+                }).addOnFailureListener(new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception e) {
+                        Log.e(ACTIVITY_TAG, "Failed signIn", e);
+                        onDisconnected(ConnectionRequest.LOG_IN);
+                    }
+                });
+    }
+
+    /**
+     * Load the latest snapshot and resolve any conflicts (if any).
+     */
+    private Task<Snapshot> loadSnapshotAndResolveConflictsTask(GoogleSignInAccount account) {
+
+        // Get the SnapshotsClient from the signed in account.
+        SnapshotsClient snapshotsClient =
+                Games.getSnapshotsClient(mActivity, account);
+
+        // Open the saved game using its name.
+        return snapshotsClient
+                .open(SAVED_GAME_FILENAME, true)
+                .addOnFailureListener(new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception e) {
+                        Log.e(ACTIVITY_TAG, "Error while opening Snapshot.", e);
+                    }
+                })
+                .continueWithTask(
+                        new Continuation<
+                                SnapshotsClient.DataOrConflict<Snapshot>,
+                                Task<Snapshot>>() {
+                            @Override
+                            public Task<Snapshot> then(
+                                    @NonNull Task<SnapshotsClient.DataOrConflict<Snapshot>> task)
+                                    throws Exception {
+                                return processSnapshotAndResolveConflictsTask(
+                                        task.getResult(),
+                                        MAX_SNAPSHOT_RESOLVE_RETRIES);
+                            }
+                        }).addOnSuccessListener(new OnSuccessListener<Snapshot>() {
+                    @Override
+                    public void onSuccess(Snapshot snapshot) {
+                        Log.i(ACTIVITY_TAG, "Successfully opened Snapshot.");
+                    }
+                });
+    }
+
+    /**
+     * Recursive function that resolves conflicts until snapshot is no longer in conflict.
+     * Will eventually give-up if snapshot is still in conflict after a supplied number of re-tries.
+     */
+    private Task<Snapshot> processSnapshotAndResolveConflictsTask(
+            final SnapshotsClient.DataOrConflict<Snapshot> result,
+            final int retryCount) {
+
+        if (!result.isConflict()) {
+            // There was no conflict, so return the result of the source.
+            TaskCompletionSource<Snapshot> source = new TaskCompletionSource<>();
+            source.setResult(result.getData());
+            return source.getTask();
+        }
+
```

Comment:
Would help debugging to log a warning `Snapshot conflict found. Attempting to resolve conflict.`